portfolio/src/main/webapp/script.js

```diff
@@ -444,7 +470,7 @@ async function displayAuth(){
  */ 
 function loginHandler(value){
     document.getElementById("authentication")
-            .innerHTML = '<a href=\"' + value.url + '\">Login</a>'
+            .innerHTML = 'Welcome to the site! <a id="loginLink" href=\"' + value.url + '\">Login Here</a>'
```

Comment:
Use the backtick (\`) operator to assign long (and potentially multi-line) strings:
```
el.innerHTML = `
  Welcome to the size!
  <a href="${value.url}">Login Here</a>`;
```

The ${} lets you inject variables directly into the string.

Reply:
Do the same in your logout handler.